README
DEMO: http://wolfslittlestore.be/labs/wolf_modal/

## Why?

Why do all lightboxes out there use a gazillion images and options when all I want is to show a larger image when I click a small one?

Features: keyboard close with Esc or X button; automatically centers in all browsers; easy to add, just add a class="lightbox" to a linked image

## Browser support

Tested in IE6, IE7, IE8, IE9, Opera 10+, FF3.5+, Safari 5, Chrome 6

## Files

* reset.css includes a CSS reset; you'll probably already use one in your project.
* modal.css includes CSS to make the modal look a bit nicer, but this CSS is not required. Ignore the file and roll your own
* modal.js & modal.min.js: the script itself
* buttons_close.png button image; the only image used

## Usage

Paste the following HTML close to the end of the <body> tag (ideally this should be the last code before the end):

<div class="modal" style="display: none;">
	<div class="mContent">
		<a href="#" class="buttonClose" title="Close this window"><strong>Close</strong> (X)</a>
		<div class="figure">
			<img src="" alt="" />
		</div>
		<p class="caption"></p>
	</div>
</div>
<div id="modalOverlay" style="display: none;">&nbsp;</div>

Link a small image to a big image: <a href="image_big.jpg"><img src="image_small.jpg" alt="" /></a>

Add a class="lightbox" to the <a>

Optional: Add alt text to the <img> for the caption

CHANGELOG
# 1.02

* Optimized PNG close button
* Added .gif close button for IE6
* Removed minified JS since it's a bitch to maintain
* The HTML gets inserted with JS now, making implementation easier
* Validated demo HTML
* Changed file names to include project name for easier recognition

# 1.01

* Added link to Github and online demo.

# 1.0

* Initial version

js/modal.js
$(document).ready(function() {

	// When clicking on an <a> with class lightbox
	$('a.lightbox').click(function(evt)
	{
		// Prevent default link behaviour
		evt.preventDefault();

		// Define contents of lightbox
		$('.figure img')
			.attr('src', $(this).attr('href'))
			.attr('alt', $(this).attr('alt'));

		// Caption = img alt tag
		$('.caption').html($(this).find('img').attr('alt'));

		// Position the box
		positionBox();

		// open lightbox
		showBox();
	});

});

function showBox() {

	$('body').addClass('modalShowing');

	// IE6/7/8 does not correctly support opacity
	if (jQuery.browser.msie) {

		// show modal box: no animation for IE
		$('#modalOverlay, .modal').show();

		// IE bugfixes
		if($.browser.msie && $.browser.version =='6.0') {
			// Persistent overlay opacity for IE6
			// Make sure modalOverlay has full height
			$('#modalOverlay').css("filter", "alpha(opacity=33)").css('height', $(document).height());
		}

	} else {
		// Simple fadeIn for the better browsers that support opacity well
		$('#modalOverlay, .modal').fadeIn("fast");
	}

	// Hide modal box when clicking outside it or on the close button
	$('#modalOverlay, .buttonClose').bind('click', function(evt) {
		// Prevent default link behaviour
		evt.preventDefault();
		//  Close modal box
		closeBox();
	});

	// Hide modal box when pressing escape button
	$('html').bind("keyup", function(evt) {
		if (evt.keyCode == 27 || evt.keyCode == 88) {
			closeBox();
		}
	});

}

function closeBox() {
	// Make sure the hide code has no effect on the modal box itself
	$('.modal').bind('click', function(evt) { evt.stopPropagation(); });

	// Too much bugs with alpha/opacity, so IE6 gets a simple hide()
	if (jQuery.browser.msie) {
		$('.modal').hide();
		$('#modalOverlay').hide();
	} else {
		// Hide modal and overlay
		$('#modalOverlay, .modal').fadeOut("fast");
	}

}

function positionBox() {
	// Centers the modal

	// initialize object references
	var oElement = $('.modal');
	var oWindow = $(window);

	// calculate offset
	var offsetLeft = parseInt((oWindow.width() - oElement.width()) / 2);
	var offsetTop = parseInt((oWindow.height() - oElement.height()) / 2);

	var modalWidth = parseInt(oElement.width());
	var modalHeight = parseInt(oElement.height());

	oElement.css('left', offsetLeft)
			.css('top', offsetTop)
			.css('width', modalWidth)
			.css('height', modalHeight)
			.css('position', 'fixed');

	// only IE6 should use absolute
	if($.browser.msie && $.browser.version =='6.0') {
		oElement.css('position', 'absolute')
		        .css('top', parseInt(offsetTop + $(window).scrollTop()));
	}
}

js/jquery.wolfmodal.js
$(document).ready(function() {

	var modalHTML = '<div class="modal" style="display: none;"><div class="mContent"><a href="#" class="buttonClose" title="Close this window"><strong>Close</strong> (X)</a><div class="figure"><img src="" alt="" /></div><p class="caption"></p></div></div><div id="modalOverlay" style="display: none\;">&nbsp\;</div>';
	$('body').append(modalHTML);

	// When clicking on an <a> with class lightbox
	$('a.lightbox').click(function(evt)
	{
		// Prevent default link behaviour
		evt.preventDefault();

		// Define contents of lightbox
		$('.figure img')
			.attr('src', $(this).attr('href'))
			.attr('alt', $(this).attr('alt'));

		// Caption = img alt tag
		$('.caption').html($(this).find('img').attr('alt'));

		// Position the box
		positionBox();

		// open lightbox
		showBox();
	});

});

function showBox() {

	// IE6/7/8 does not correctly support opacity
	if (jQuery.browser.msie) {

		// show modal box: no animation for IE
		$('#modalOverlay, .modal').show();

		// IE bugfixes
		if($.browser.msie && $.browser.version =='6.0') {
			// Persistent overlay opacity for IE6
			// Make sure modalOverlay has full body height
			$('#modalOverlay').css("filter", "alpha(opacity=33)").css('height', $('body').height());
		}

	} else {
		// Simple fadeIn for the better browsers that support opacity well
		$('#modalOverlay, .modal').fadeIn("fast");
	}

	// Hide modal box when clicking outside it or on the close button
	$('#modalOverlay, .buttonClose').bind('click', function(evt) {
		// Prevent default link behaviour
		evt.preventDefault();
		//  Close modal box
		closeBox();
	});

	// Hide modal box when pressing escape button
	$('html').bind("keyup", function(evt) {
		if (evt.keyCode == 27 || evt.keyCode == 88) {
			closeBox();
		}
	});

}

function closeBox() {
	// Make sure the hide code has no effect on the modal box itself
	$('.modal').bind('click', function(evt) { evt.stopPropagation(); });

	// Too much bugs with alpha/opacity, so IE6 gets a simple hide()
	if (jQuery.browser.msie) {
		$('#modalOverlay, .modal').hide();
	} else {
		// Hide modal and overlay
		$('#modalOverlay, .modal').fadeOut("fast");
	}

}

function positionBox() {

	// initialize object references
	var oElement = $('.modal');
	var oWindow = $(window);

	// calculate offset
	var offsetLeft = parseInt((oWindow.width() - oElement.width()) / 2);
	var offsetTop = parseInt((oWindow.height() - oElement.height()) / 2);

	var modalWidth = parseInt(oElement.width());
	var modalHeight = parseInt(oElement.height());

	oElement.css('left', offsetLeft)
			.css('top', offsetTop)
			.css('width', modalWidth)
			.css('height', modalHeight)
			.css('position', 'fixed');

	// IE6 should use position: absolute; since fixed is not supported
	if($.browser.msie && $.browser.version =='6.0') {
		oElement.css('position', 'absolute')
		        .css('top', offsetTop + oWindow.scrollTop());
	}

}